crates/oxc_semantic/src/symbol/mod.rs

```diff
@@ -13,20 +15,22 @@ pub use self::{
     reference::{Reference, ReferenceFlag},
     table::SymbolTable,
 };
+use crate::node::AstNodeId;
 
 #[derive(Debug)]
 pub struct Symbol {
     id: SymbolId,
     name: Atom,
     span: Span,
     flags: SymbolFlags,
+    references: RefCell<Vec<AstNodeId>>,
 }
 
 #[cfg(all(target_arch = "x86_64", target_pointer_width = "64"))]
 #[test]
 fn symbol_size() {
     use std::mem::size_of;
-    assert_eq!(size_of::<Symbol>(), 48);
+    assert_eq!(size_of::<Symbol>(), 80);
 }
 
```

Comment:
Should we move references away to something like a `HashMap<SymbolId, Vec<AstNodeId>>` so that the Symbol structure's size doesn't grow?

Reply:
`RefCell` can be removed, `Vec` can become `ThinkVec`, we can also try "struct of arrays". But let's work on optimisations in another PR.